iris-mpc/src/bin/server.rs

```diff
@@ -695,26 +697,44 @@ async fn server_main(config: Config) -> eyre::Result<()> {
     let is_ready_flag = Arc::new(AtomicBool::new(false));
     let is_ready_flag_cloned = Arc::clone(&is_ready_flag);
 
+    #[derive(Serialize, Deserialize)]
+    struct ReadyProbeResponse {
+        image_name: String,
+        uuid:       String,
+    }
+
     let _health_check_abort = background_tasks.spawn({
         let uuid = uuid::Uuid::new_v4().to_string();
+        let ready_probe_response = ReadyProbeResponse {
+            image_name: config.image_name.clone(),
+            uuid,
+        };
+        let serialized_response =
+            bincode::serialize(&ready_probe_response).expect("Serialization failed");
         async move {
             // Generate a random UUID for each run.
-            let app = Router::new()
-                .route("/health", get(move || async move { uuid.to_string() }))
-                .route(
-                    "/ready",
-                    get({
-                        // We are only ready once this flag is set to true.
-                        let is_ready_flag = Arc::clone(&is_ready_flag);
-                        move || async move {
-                            if is_ready_flag.load(Ordering::SeqCst) {
-                                "ready".into_response()
-                            } else {
-                                StatusCode::SERVICE_UNAVAILABLE.into_response()
+            let app =
+                Router::new()
+                    .route(
+                        "/health",
+                        get(move || async move {
+                            general_purpose::STANDARD.encode(&serialized_response)
```

Comment:
a bit awkward that we base64 the binary serialization, why not json serde json it?

Reply:
Sure! Wanted to take advantage of some more rust quirks but makes sense to use JSON